atd-vzd/migrations/migration_noncr3_location_2023_08_02--1648.sql

```diff
@@ -0,0 +1,37 @@
+-- 1. Add triggers to update location_id on insert and update
+CREATE TRIGGER update_noncr3_location_on_insert
+AFTER INSERT ON atd_apd_blueform
+FOR EACH ROW
+WHEN (NEW.latitude IS NOT NULL AND NEW.longitude IS NOT NULL)
```

Comment:
im wondering whether `position` is what should be checked here since thats whats used later in the trigger function?

Reply:
this is a good point. It also makes me wonder if using [ST_Equals](https://postgis.net/docs/en/ST_Equals.html) for the update trigger would be the right way to go about comparing `NEW` to `OLD`. I'm going to look into this. 🙏

Reply:
@johnclary @frankhereford this is what I was talking about in planning - any experience with this?

Reply:
i'm having trouble finding the docs that support this, but my understanding is that for point geometries you can test with an old fashion equality operator. e.g., `NEW.position != OLD.position`. 

for sure `ST_Equals` will get the job done for you though!

as an aside, i suspect that testing on the two numeric columns would be the most performant (by probably a very thin margin). e.g. `NEW.x != OLD.x AND NEW.y != OLD.y`

Reply:
finally coming back to this! I'm going to use a bit of both of y'all's advice and stick with using lat and lon with `IS DISTINCT FROM`. Thanks!

Reply:
@mddilley wrote:

> @johnclary @frankhereford this is what I was talking about in planning - any experience with this?

Generally, I think sticking with the postGIS functions are the way to go when operating on GIS data. There are just so many little gotchas, none-the-least-of-which is that you're comparing equality of floating point numbers if you go at it via the latitude and longitude. `0.999… == 1`, `sqrt(2)**2 != 2`, etc. I always want to let the gis libraries deal with all that minutia if I can.

That said, postGIS is going to have overloaded operators based on the input type, so != will become the GIS functions that test for non-equality if the fields you're `!=`ing are GIS data. I think you're good with something like `NEW.position != OLD.position`.